vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -311,6 +313,12 @@ func newEventRecorder(kubeClient kube_client.Interface) record.EventRecorder {
 	eventBroadcaster.StartLogging(klog.V(4).Infof)
 	if _, isFake := kubeClient.(*fake.Clientset); !isFake {
 		eventBroadcaster.StartRecordingToSink(&clientv1.EventSinkImpl{Interface: clientv1.New(kubeClient.CoreV1().RESTClient()).Events("")})
+	} else {
+		eventBroadcaster.StartRecordingToSink(&clientv1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
 	}
-	return eventBroadcaster.NewRecorder(scheme.Scheme, apiv1.EventSource{Component: "vpa-updater"})
+
+	vpascheme := scheme.Scheme
+	corescheme.AddToScheme(vpascheme)
```

Comment:
Check for error?
```
if err := corescheme.AddToScheme(vpascheme); err != nil {
        klog.Fatalf("Error adding core scheme: %v", err)
    }
```

Reply:
Whoops! Thanks for catching that. Fixed in https://github.com/kubernetes/autoscaler/pull/7413/commits/f6ada00f2cb1cc7f7819f09af87cee9a6ad1c8c7